src/SerilogWeb.Classic/Classic/ApplicationLifecycleModule.cs

```diff
@@ -133,31 +136,56 @@ public static LogEventLevel FormDataLoggingLevel
         /// <param name="context">An <see cref="T:System.Web.HttpApplication"/> that provides access to the methods, properties, and events common to all application objects within an ASP.NET application </param>
         public void Init(HttpApplication context)
         {
-            context.LogRequest += LogRequest;
-            context.Error += Error;
-        }
-
-        static void LogRequest(object sender, EventArgs e)
-        {
-            if (!_isEnabled) return;
-
-            var request = HttpContextCurrent.Request;
-            if (request == null)
-                return;
+            context.BeginRequest += (sender, args) =>
+            {
+                if(_isEnabled)
+                {
+                    context.Context.Items[StopWatchKey] = Stopwatch.StartNew();
+                }                
+            };
 
-            Logger.Write(_requestLoggingLevel, "HTTP {Method} for {RawUrl}", request.HttpMethod, request.RawUrl);
-            if (ShouldLogRequest())
+            context.EndRequest += (sender, args) =>
             {
-                var form = request.Unvalidated.Form;
-                if (form.HasKeys())
+                if (_isEnabled)
                 {
-                    var formData = form.AllKeys.SelectMany(k => (form.GetValues(k) ?? new string[0]).Select(v => new { Name = k, Value = FilterPasswords(k, v) }));
-                    Logger.Write(_formDataLoggingLevel, "Client provided {@FormData}", formData);
+                    Stopwatch stopwatch = (Stopwatch)context.Context.Items[StopWatchKey];
```

Comment:
I'm sorry, I don't think I fully understood what you were referring to. I had to use `context.Context` to get access to the `Items` dictionary to stash the stopwatch. 

I added a check to make sure there is an available `HttpApplication.Context`, and renamed the HttpApplication variable to application from context since it was confusing.